view/frontend/templates/js/replacejs.phtml

```diff
@@ -1714,16 +1717,48 @@ if (!$block->isSaveCartInSections()) { ?>
                 if (waitingForResolvingPromises && (localCallNumber === countConfigureCalls)) {
                     customerData.reload(['boltcart'], true);
                 }
-            }, 5000);
+            }, 10000);
         }
 
         var invalidateBoltCart = function() {
             customerData.reload(['boltcart'], true);
             callConfigureWithPromises();
         }
 
+        var magentoCart;
+        var boltCartDataID;
+
+        var invalidateBoltCartIfOutdated = function() {
+            setTimeout(function () {
+                if (magentoCart === undefined || magentoCart.data_id === undefined || boltCartDataID === undefined) {
+                    return;
+                }
+                if (magentoCart.data_id > boltCartDataID) {
+                    invalidateBoltCart();
+                    return;
+                }
+                if (magentoCart.data_id == boltCartDataID) {
```

Comment:
It doesn't look like a possible real case, when magento cart and bolt cart have the same timestamp but bolt cart generated before and it is outdated.
But just in case I decided to add this check.